lib/db/sprint.ts

```diff
@@ -0,0 +1,96 @@
+import prisma from "./prisma";
+import { Sprint } from "@prisma/client";
+
+/**
+ * Retrieves a single sprint based on provided parameters.
+ *
+ * @export
+ * @async
+ * @param {Sprint} params
+ * @returns {Promise<Sprint | null>} The retrieved sprint object or null
+ */
+export async function getSprint(
+  params: Partial<Sprint>
+): Promise<Sprint | null> {
+  const sprint = await prisma.sprint.findFirst({
+    where: {
+      ...params,
+    },
+    include: { tasks: true },
+  });
+  return sprint;
+}
+
+/**
+ * Retrieves multiple sprints based on provided parameters.
+ *
+ * @export
+ * @async
+ * @param {Sprint} params
+ * @returns {Promise<Sprint[]>} An array of retrieved sprint objects.
+ */
+export async function getSprints(params: Partial<Sprint>): Promise<Sprint[]> {
+  const sprints = await prisma.sprint.findMany({
+    where: {
+      ...params,
+    },
+    include: { tasks: true },
+  });
+  return sprints;
+}
+
+/**
+ * Creates a new sprint.
+ *
+ * @export
+ * @async
+ * @param {Sprint} params Information about the sprint to create.
+ * @returns {Promise<Sprint>} The created sprint object.
+ */
+export async function createSprint(params: Partial<Sprint>): Promise<Sprint> {
+  const sprint = await prisma.sprint.create({
+    data: {
+      project: { connect: { id: params.project_id } },
+      title: params.title,
+      start_date: params.start_date,
+      end_date: params.end_date,
+      planned_capacity: params.planned_capacity,
```

Comment:
Looks like you forgot to use spread syntax here? I do see that you linked the `project_id` manually -- is this necessary, or would passing `project_id` as part of the `Sprint` object suffice to create the relation?

Reply:
@winstuhn Can I get your feedback here?

Reply:
I think it requires manually linking the related entities. As a result, for the create functions I cannot use the spread syntax. I'll keep looking for a fix for this, however.